internal/service/public_methods.go

```diff
@@ -0,0 +1,150 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ElenaMikhailovskaya/go_final_project/internal/models"
+	"time"
+)
+
+func (s *Server) AddTask(task models.TaskRequest) (int, error) {
+
+	var date string
+	var nextDate string
+	var newTaskId int
+	var err error
+
+	// проверка даты
+	if task.Date == "" {
+		date = time.Now().Format(models.DateFormat)
+	} else {
+		_, err = time.Parse(models.DateFormat, task.Date)
+		if err != nil {
+			return 0, errors.New("Invalid date format")
+		}
+		date = task.Date
+	}
+
+	// проверка параметра повторения
+	if task.Repeat == "" {
+		nextDate = time.Now().Format(models.DateFormat)
+	} else {
+		now := time.Now().Format(models.DateFormat)
+		nextDate, err = s.NextDate(now, date, task.Repeat, false)
+		if err != nil {
+			return 0, err
```

Comment:
Для простоты чтения ошибки ее можно обернуть
```suggestion
			return 0, fmt.Errorf("error in NextDate: %w",err)
```

Reply:
исправила